tests/Microsoft.Identity.Test.Unit/WwwAuthenticateParametersTests.cs

```diff
@@ -56,6 +59,86 @@ public void CreateWwwAuthenticateResponse(string resource, string authorizationU
             Assert.IsNull(authParams.Error);
         }
 
+        [TestMethod]
+        [DataRow("AuthScheme", "" )]
+        [DataRow("AuthScheme", "realm = someRealm")]
+        [DataRow("AuthScheme", "token68")]
+        [DataRow("AuthScheme", "realm = someRealm token68")]
+        [DataRow("AuthScheme", "auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "realm = someRealm auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "token68 auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        [DataRow("AuthScheme", "realm = someRealm token68 auth-param1=token1, auth-param2=token2, auth-param3=token3")]
+        public void EnsureProperlyFormattedHeadersDoNotFail(string scheme, string values)
+        {
+            // Arrange
+            HttpResponseMessage httpResponse = new HttpResponseMessage(HttpStatusCode.Unauthorized);
+            httpResponse.Headers.Add(WwwAuthenticateHeaderName, $"{scheme} {values}");
+
+            // Act
+            var authParams = WwwAuthenticateParameters.CreateFromAuthenticationHeaders(httpResponse.Headers, scheme);
+
+            // Assert
+            //if (scheme == "NTLM")
+            //{
+            //    Assert.AreEqual(scheme, authParams.AuthScheme);
+            //    Assert.AreEqual(values, authParams.RawParameters[scheme]);
+            //}
+            //else
+            //{
+            //    Assert.AreEqual(3, authParams.RawParameters.Count);
+            //    Assert.AreEqual("WindowsLive", authParams.RawParameters["realm"]);
+            //    Assert.AreEqual("MBI_SSL", authParams.RawParameters["policy"]);
+            //    Assert.AreEqual("ssl.live-tst.net", authParams.RawParameters["siteId"]);
+            //}
+        }
+
+        [TestMethod]
+        [DataRow("WLID1.0", "realm=WindowsLive, policy=MBI_SSL, siteId=\"ssl.live-tst.net\"")]
+        [DataRow("NTLM", "dG9rZW42OA==")]

```

Comment:
As before, RawParameters should probably not have {"NTLM": "dG9.."} as it does not reflect the state.